sw/airborne/subsystems/gps.c

```diff
@@ -355,10 +357,15 @@ struct UtmCoor_f utm_float_from_gps(struct GpsState *gps_s, uint8_t zone)
     utm.north = gps_s->utm_pos.north / 100.0f;
     utm.alt = gps_s->utm_pos.alt / 1000.f;
   }
-  else {
+  else if (bit_is_set(gps_s->valid_fields, GPS_VALID_POS_LLA_BIT)){
+    /* Recompute UTM coordinates in this zone */
     struct UtmCoor_i utm_i;
-    utm_i.zone = zone;
     utm_of_lla_i(&utm_i, &gps_s->lla_pos);
+    if (bit_is_set(gps_s->valid_fields, GPS_VALID_HMSL_BIT)) {
+      utm.alt = gps.hmsl/1000.;
+    } else {
+      utm.alt -= NAV_MSL0/1000.;
```

Comment:
that doesn't make any sense... it is a fixed value
If HMSL is not valid in GPS, it should fall back to calculation from ellipsoid alt and geoid_separation